### PR TITLE
Xtokens call restrict

### PR DESCRIPTION
**Pull request by wangjj9219**

### CODE REVIEW

File: modules/extrinsic-restrict/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,336 @@
...
+	pub type Hihi<T: Config> = StorageMap<_, Twox64Concat, CurrencyId, XtokensTransferLimit, ValueQuery>;
+
+	#[pallet::pallet]
+	#[pallet::without_storage_info]
+	pub struct Pallet<T>(_);
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<T::BlockNumber> for Pallet<T> {
+		fn on_initialize(now: T::BlockNumber) -> Weight {
+			if now
```
Comment (xlc) [inline modules/extrinsic-restrict/src/lib.rs:111]: we shouldn't need to clear the limit. just store the last update time is enough

File: modules/extrinsic-restrict/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,52 @@
+[package]
+name = "module-extrinsic-restrict"
```
Comment (xlc) [inline modules/extrinsic-restrict/Cargo.toml:2]: just call it `module-rate-limit`

File: modules/extrinsic-restrict/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,336 @@
...
+mod mock;
+mod tests;
+pub mod weights;
+
+#[frame_support::pallet]
+pub mod module {
+	use super::*;
+
+	#[derive(PartialEq, Eq, Clone, Encode, Decode, RuntimeDebug, TypeInfo, Default)]
+	pub struct XtokensTransferLimit {
```
Comment (xlc) [inline modules/extrinsic-restrict/src/lib.rs:61]:
```suggestion
	pub struct RateLimit {
```

no need to reference xtokens in this pallet

**Comment by wangjj9219** [comment]
new impl #2419

**Pull request closed without merging**